### PR TITLE
user model and vehicle model created

### PR DESCRIPTION
**Pull request by shubham0809200**

### CODE REVIEW

File: src/main/java/com/capool/backend/model/VehicleModel.java [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+
+    @Column(name = "VEHICLE_TYPE", nullable = false)
+    private String vehicleType;
+
+    @Column(name = "VEHICLE_BRAND", nullable = false)
+    private String vehicleBrand;
+
+    @Id
+    @Column(name = "VEHICLE_NUMBER", nullable = false)
+    private String vehicleNumber;
```
Comment (ayushkumar1610) [inline src/main/java/com/capool/backend/model/VehicleModel.java:21]: Need to map with UserModel: Many to One

File: src/main/java/com/capool/backend/model/UserModel.java [hunk 2]
```diff
@@ -0,0 +1,189 @@
...
+
+    @Column(name = "PASSWORD", nullable = false)
+    private String password;
+
+    @Column(name = "PHONE_NUMBER")
+    private int phone;
+    private String country;
+
+    private boolean hasVehicle;
+    private VehicleModel vehicleList;
```
Comment (ayushkumar1610) [inline src/main/java/com/capool/backend/model/UserModel.java:35]: vehicleList need to mapped with VehicleModel: One to many

File: src/main/java/com/capool/backend/model/UserModel.java [hunk 3]
```diff
@@ -0,0 +1,189 @@
...
+
+    @Column(name = "GENDER", nullable = true)
+    private String gender;
+
+    private String currentLocation;
+
+    @Column(name = "ROLE", nullable = false)
+    private String role;
```
Comment (ayushkumar1610) [inline src/main/java/com/capool/backend/model/UserModel.java:43]: set role data type as per Enum

File: src/main/java/com/capool/backend/model/UserModel.java [hunk 4]
```diff
@@ -0,0 +1,189 @@
...
+        this.hasVehicle = hasVehicle;
+    }
+
+    public VehicleModel getVehicleList() {
+        return vehicleList;
+    }
+
+    public void setVehicleList(VehicleModel vehicleList) {
+        this.vehicleList = vehicleList;
+    }
```
Comment (ayushkumar1610) [inline src/main/java/com/capool/backend/model/UserModel.java:132]: Set data type as ```List<VehicleModel>```

File: src/main/java/com/capool/backend/model/UserModel.java [hunk 5]
```diff
@@ -0,0 +1,189 @@
...
+        this.currentLocation = currentLocation;
+    }
+
+    public String getRole() {
+        return role;
+    }
+
+    public void setRole(String role) {
+        this.role = role;
+    }
```
Comment (ayushkumar1610) [inline src/main/java/com/capool/backend/model/UserModel.java:156]: Modify datatype as Enum

File: src/main/java/com/capool/backend/model/VehicleModel.java [hunk 6]
```diff
@@ -0,0 +1,124 @@
...
+    private String vehicleModel;
+
+    @Column(name = "VEHICLE_COLOR", nullable = true)
+    private String vehicleColor;
+
+    @Column(name = "VEHICLE_CAPACITY", nullable = false)
+    private String vehicleCapacity;
+
+    @Column(name = "VEHICLE_OWNER", nullable = false)
+    private String vehicleOwner;
```
Comment (ayushkumar1610) [inline src/main/java/com/capool/backend/model/VehicleModel.java:33]: Modify datatype as UserModel and map as One to One
**_Suggestion_**: Should we allow multiple owners of a vehicle? if so then datatype will be ```List<UserModel>``` with One to Many mapping

File: src/main/java/com/capool/backend/model/VehicleModel.java [hunk 7]
```diff
@@ -0,0 +1,124 @@
...
+
+    public VehicleModel() {
+        super();
+    }
```
Comment (ayushkumar1610) [inline src/main/java/com/capool/backend/model/VehicleModel.java:37]: Put constructor above all methods

File: src/main/java/com/capool/backend/model/VehicleModel.java [hunk 8]
```diff
@@ -0,0 +1,124 @@
...
+        this.vehicleCapacity = vehicleCapacity;
+    }
+
+    public String getVehicleOwner() {
+        return vehicleOwner;
+    }
+
+    public void setVehicleOwner(String vehicleOwner) {
+        this.vehicleOwner = vehicleOwner;
+    }
```
Comment (ayushkumar1610) [inline src/main/java/com/capool/backend/model/VehicleModel.java:114]: Modify datatype as UserModel
Or if multiple owner then ```List<UserModel>```

File: src/main/java/com/capool/backend/model/VehicleModel.java [hunk 9]
```diff
@@ -0,0 +1,124 @@
...
+
+    @Override
+    public String toString() {
+        return String.format(
+                "VehicleModel [ id=%d, vehicleType=%s, vehicleNumber=%s, vehicleModel=%s, vehicleColor=%s, vehicleCapacity=%s, vehicleOwner=%s]",
+                id, vehicleType, vehicleNumber, vehicleModel, vehicleColor, vehicleCapacity, vehicleOwner);
+
+    }
```
Comment (ayushkumar1610) [inline src/main/java/com/capool/backend/model/VehicleModel.java:122]: remove vehicle owner

File: src/main/java/com/capool/backend/service/user/UserService.java [hunk 10]
```diff
@@ -0,0 +1,5 @@
+package com.capool.backend.service.user;
+
+public class UserService {
+    
+}
```
Comment (ayushkumar1610) [inline src/main/java/com/capool/backend/service/user/UserService.java:5]: Make it interface

File: src/main/java/com/capool/backend/service/user/UserServiceImpl.java [hunk 11]
```diff
@@ -0,0 +1,5 @@
+package com.capool.backend.service.user;
+
+public class UserServiceImpl {
+    
+}
```
Comment (ayushkumar1610) [inline src/main/java/com/capool/backend/service/user/UserServiceImpl.java:5]: implements UserService

File: src/main/java/com/capool/backend/model/UserModel.java (hunk 2 above)
Comment (ayushkumar1610) [reply src/main/java/com/capool/backend/model/UserModel.java:35]: done

File: src/main/java/com/capool/backend/model/UserModel.java (hunk 4 above)
Comment (ayushkumar1610) [reply src/main/java/com/capool/backend/model/UserModel.java:132]: done

File: src/main/java/com/capool/backend/model/VehicleModel.java (hunk 1 above)
Comment (team-carpool) [reply src/main/java/com/capool/backend/model/VehicleModel.java:21]: not required

File: src/main/java/com/capool/backend/model/UserModel.java (hunk 3 above)
Comment (team-carpool) [reply src/main/java/com/capool/backend/model/UserModel.java:43]: done

File: src/main/java/com/capool/backend/model/UserModel.java (hunk 5 above)
Comment (team-carpool) [reply src/main/java/com/capool/backend/model/UserModel.java:156]: done

File: src/main/java/com/capool/backend/model/VehicleModel.java (hunk 6 above)
Comment (team-carpool) [reply src/main/java/com/capool/backend/model/VehicleModel.java:33]: mapped as ManyToOne
many vehicle with one user

File: src/main/java/com/capool/backend/model/VehicleModel.java (hunk 8 above)
Comment (team-carpool) [reply src/main/java/com/capool/backend/model/VehicleModel.java:114]: done

File: src/main/java/com/capool/backend/model/VehicleModel.java (hunk 9 above)
Comment (team-carpool) [reply src/main/java/com/capool/backend/model/VehicleModel.java:122]: done

**Pull request merged**